lora-codec-lansitec/readme.md

```diff
@@ -0,0 +1,514 @@
+            ObjectMapper mapper = new ObjectMapper();
+            String payload = null;
+            try {
+                if (lpwanEncoderInputData.getCommandName().equals(POSITION_REQUEST)) {
+                    payload = "A1FF";
+                } else if (lpwanEncoderInputData.getCommandName().equals(REGISTER_REQUEST)) {
+                    payload = "A2FF";
+                } else if (lpwanEncoderInputData.getCommandName().equals(DEVICE_REQUEST)) {
+                    payload = "A3FF";
+                } else if (lpwanEncoderInputData.getCommandName().equals(SET_CONFIG)) {
+                    JsonNode params = mapper.readTree(lpwanEncoderInputData.getCommandData());
+                    ByteBuffer buffer = ByteBuffer.allocate(4).order(ByteOrder.BIG_ENDIAN);
+                    byte breakpoint = params.get("breakpoint").asBoolean() ? (byte) 8 : 0;
+                    byte selfadapt = params.get("selfadapt").asBoolean() ? (byte) 4 : 0;
+                    byte oneoff = params.get("oneoff").asBoolean() ? (byte) 2 : 0;
+                    byte alreport = params.get("alreport").asBoolean() ? (byte) 1 : 0;
+                    buffer.put((byte) ((byte) 0x90 | (byte) breakpoint | (byte) selfadapt | (byte) oneoff | (byte) alreport));
+                    buffer.putShort((short) params.get("pos").asInt());
+                    buffer.put((byte) params.get("hb").asInt());
+                    payload = BaseEncoding.base16().encode(buffer.array());
+                }
+
+                encoderResult = new LpwanEncoderResult(payload, 20);
+                encoderResult.setSuccess(true);
+                encoderResult.setMessage("Successfully Encoded the payload");
+            } catch (IOException e) {
+                e.printStackTrace();
+                encoderResult = new LpwanEncoderResult();
+                encoderResult.setSuccess(false);
+                encoderResult.setMessage("Encoding Payload Failed");
+            }
+        }
+        return encoderResult;
+    }
+```
+
+## Deploying the example codec microservice
+
+In order to build and deploy the sample codec microservice, follow [Microservice Guide](http://www.cumulocity.com/guides/microservice-sdk/java/)
+
+Clone this repository first. Next, build the microservice using `mvn clean install`. The build will create a zip file of the decoder microservice.
+
+In the next step, deploy the microservice using Cumulocity IoT UI. Once the decoder microservice has been deployed, wait a couple of minutes in order to allow Cumulocity IoT to discover the new decoder. Then, open Device Management UI. Inside Device Protocols, you should now see the device types with type as 'lpwan' created by codec microservice. Map one of the device types to the LPWAN device and send uplink message to see the respective measurements 
```

Comment:
```suggestion
In the next step, deploy the microservice using the Cumulocity IoT UI. Once the decoder microservice has been deployed, it can take a couple of minutes Cumulocity IoT to discover the new decoder. Then, open the device management application. Inside device protocols, you should now see the device types with type as 'lpwan' created by codec microservice. Map one of the device types to the LPWAN device and send an uplink message to see the respective measurements 
```